### PR TITLE
Generate API docs

### PR DESCRIPTION
**Pull request by gitlawr**
Related issue: https://github.com/harvester/harvester/issues/890

With this change, `go generate` also creates openAPI specs.

### CODE REVIEW

File: scripts/known-api-rule-violations.txt [hunk 1]
```diff
@@ -0,0 +1,376 @@
+API rule violation: list_type_missing,github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1,AuthenticationModesResponse,Modes
```
Comment (guangbochen) [inline scripts/known-api-rule-violations.txt:1]: why are those fields have list_type_missing?

File: swagger.json [hunk 2]
```diff
@@ -0,0 +1,13349 @@
+{
```
Comment (guangbochen) [inline swagger.json]: seems added by accdient?

**Review by guangbochen — COMMENTED** [review]
btw is `pkg/apis/harvesterhci.io/v1beta1/openapi_generated.go` needed, this seems to be a really large file, thanks.

File: scripts/known-api-rule-violations.txt (hunk 1 above)
Comment (gitlawr) [reply scripts/known-api-rule-violations.txt:1]: See https://github.com/kubernetes/kube-openapi/issues/175#issuecomment-546504117

**Comment by gitlawr** [comment]
The `genswagger` code uses `openapi_generated.go` as the dependency. If we remove the `openapi_generated.go`, the linter checks will fail.

**Comment by gitlawr** [comment]
Updated @guangbochen

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**